### ISSUE TITLE
Multiple Set location... do not work

### ISSUE THREAD

**Issue by deejam** [transmission/transmission#5993]
### What is the issue?

Changing the location of multiple torrents at the same time do not work.

How to reproduce:
1. Select more than one torrent
2. Right click and select "Set location..."
3. Change the location
4. Click "OK"

Expected behavior: New location is applied to all selected torrents.
Current behavior: New location is only applied to the torrent that was under the mouse cursor when right-clicking.

### Which application of Transmission?

Qt app on Windows

### Which version of Transmission?

4.0.2 (2a57b17031)

**Comment by GaryElshaw** [comment]
I think that's by design because each torrent will either be its own stand-alone file or folder.

**Comment by deejam** [comment]
It is clearly a bug:

1. It is possible to select multiple torrents, right click and apply a new location.
If changing the download location of multiple torrents at the same time should not be possible, then "Set location..." should be greyed out when more than one torrent is selected.
2. There is no technical reason limiting applying a new location to several torrents at the same time.
The parent directory location is set which can be done for multiple torrents at the same time.
The exact same procedure is performed when selecting several torrents at the same time as when individually selecting torrents. The only difference is how many torrents are selected (one or multiple).

**Comment by killemov** [comment]
Looks like a bug to me. The [RPC Spec](https://github.com/transmission/transmission/blob/main/docs/rpc-spec.md#36-moving-a-torrent) clearly allows for multiple torrents to be relocated. Weird thing is that [`torrent-set-location`](https://github.com/killemov/Shift/blob/ac0c3dd2633c517ac5aaba435e7d56f86ad21aed/shift.js#L3372) ha(s/d) to be used instead of just changing `downloadDir` for a single torrent.

**Comment by killemov** [comment]
@tearfur This is probably in libtransmission scope instead of just qt.

**Comment by tearfur** [comment]
@killemov Are you throwing out blind guesses? If you are then I politely ask you to stop: You never tell us where you got your ideas, it is not your first time, and it is only mucking up the discussion.

I'd appreciate it if we our conversation has researched facts, test results, code snippets, suggestions. It's fine even if you are unsure about your findings, it is already a great service to everyone in that you gave out your free time for trying it.

Back to the bug: This works properly in the WebUI, so I can already be sure there's nothing wrong with the RPC API nor libtransmission implementation details. And since the C API `tr_torrentSetLocation()` literally has the same code as RPC `torrent-set-location` for a single torrent, I am pretty sure there's nothing wrong with the C API as well.

**Comment by killemov** [comment]
I have had a similar issue popup a loooong time ago around when the method `torrent-set-location` was introduced and apparently erroneously linked that to this.

**Comment by killemov** [comment]
> You are obviously knowledgeable. In the 2 years i have been a contributor to the project, and i may have missed it, i haven't seen you contribute a pr. Instead,

Correct. At most I have done a tiny bit of code review and hinting at the use of the FileReader API to Jordan, which I used in Shift from the start. This did lead to the eventual removal of the old upload scheme.

> i often feel like this forum is a means to either criticise the project, or to market Shift. That's fine, but if you see something wrong, please help fix it.

I do criticize the project, mainly about decisions made in the pre-GitHub era. For example, the trackerAdd, trackerRemove and trackerReplace were implemented with single torrent use in mind. This was not documented. In fact, the implementation was so fragile that the team decided to go with a shared tracker list instead. I don't know when you became a Transmission user but the project was in zombie mode for several years. Many improvements were offered as patches in Trac and were ignored because of "reasons". I believe @mikedld and @ckerr pulled the project from the brink of death and I'm grateful for that. I do reference [Shift](https://github.com/killemov/Shift) whenever I believe something was/will be implemented there (better).

I saw your typo (crticise) and fixed it. :sweat_smile: But seriously my c/c++ is rusty at best and obviously I have a very different philosophy about the web-UI design and implementation.

Apologies to @deejam for even the appearance of hijacking his issue.

Also note that the original, longer, message I wrote got lost in the GitHub void because I got signed out involuntarily during composing.
